Disher/View Controllers/RecipesViewController.m

```diff
@@ -484,6 +354,172 @@ - (void) sortIngredients {
+        imageLink = recipe[@"image"];
+        mealID = [NSString stringWithFormat:@"%@", recipe[@"id"]];
+        source = @"Spoonacular";
+        if (((NSArray *)recipe[@"cuisines"]).count) {
+            cuisine = recipe[@"cuisines"];
+            for (NSString *cuisine in recipe[@"cuisines"]) {
+                [self.cuisines addObject:cuisine];
+            }
+        }
+        else {
+            cuisine = @[@"Unknown"];
+            [self.cuisines addObject:@"Unknown"];
+        }
+    }
+    Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:source withID:mealID withCuisine:cuisine];
+    [self.tableViewRecipes addObject:newRecipe];
+    self.unfilteredTableViewRecipes = [self.tableViewRecipes copy];
+    [self refreshData];
+}
+
+
+- (void) handleIngredientSearch:(NSString *)type withMeals:(NSArray *)meals withInput:(NSString *) input {
+    NSString *recipeName;
+    NSString *imageLink;
+    NSString *mealID;
+    NSString *source;
+    if ([type isEqualToString:@"Spoonacular"]) {
+        for (NSDictionary *meal in meals) {
+            recipeName = meal[@"title"];
+            imageLink = meal[@"image"];
+            mealID = [NSString stringWithFormat:@"%@", meal[@"id"]];
+            source = @"Spoonacular";
+            [Recipe getRecipeInfo:mealID withSource:source withCompletion:^(NSDictionary * _Nonnull recipeInformation) {
+                NSArray *cuisines = [self handleCuisineFilteringWithDictionary:recipeInformation withSource:@"Spoonacular"];
+                Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:source withID:mealID withCuisine:cuisines];
+                [self.tableViewRecipes addObject:newRecipe];
+                [self handleIngredientMatching:meal withMealID:mealID withSource:@"Spoonacular" withInput:input];
+                self.unfilteredTableViewRecipes = [self.tableViewRecipes copy];
+                [self refreshData];
+            }];
+        }
+    }
+    else if ([type isEqualToString:@"TheMealDB"]) {
+        for (NSDictionary *meal in meals) {
+            recipeName = meal[@"strMeal"];
+            imageLink = meal[@"strMealThumb"];
+            mealID = meal[@"idMeal"];
+            source = @"TheMealDB";
+            [Recipe getRecipeInfo:mealID withSource:@"TheMealDB" withCompletion:^(NSDictionary * _Nonnull recipeInformation) {
+                NSArray *cuisine = [self handleCuisineFilteringWithDictionary:recipeInformation withSource:@"TheMealDB"];
+                Recipe *newRecipe = [Recipe initWithRecipe:recipeName withURL:imageLink withSource:source withID:mealID withCuisine:cuisine];
+                [self handleIngredientMatching:recipeInformation withMealID:mealID withSource:@"TheMealDB" withInput:input];
+                [self.tableViewRecipes addObject:newRecipe];
+                self.unfilteredTableViewRecipes = [self.tableViewRecipes copy];
+                [self refreshData];
```

Comment:
This could also be another helper method so we don't have as much code reuse from 425-431